### PR TITLE
support comments in empty Go maps/lists

### PR DESCRIPTION
**Pull request by josharian**
I realized this was easy. Silly me.

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [-] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [-] I have not broken the cheatsheet

### CODE REVIEW

**Review by pokey — COMMENTED** [review]
what happens with

```go
T{
   /* foo */
   0,
   /* foo */
}
```

Put another way, does the `*` on `(comment)*` require the comments to be adjacent? If not, then this example would be treated as a map, when it should not

**Comment by josharian** [comment]
> Put another way, does the * on (comment)* require the comments to be adjacent?

Empirically, yes. Added tests showing with and without.

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/go/changeEveryMapFile2.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    target:
+      type: primitive
+      modifiers:
+        - type: everyScope
+          scopeType: {type: map}
+        - type: containingScope
+          scopeType: {type: document}
+  usePrePhraseSnapshot: true
+initialState:
+  documentContents: _ = []T{T{ /* foo */ /*0,*/ /* foo */ }, T{}}
```
Comment (pokey) [inline packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/go/changeEveryMapFile2.yml:16]: I'm a bit confused by this test. I was imagining you'd put my example in a file, say "cursorless record error", put your cursor next to the structure, and say "change map". If it's an error, then we're good

Comment (josharian) [reply packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/go/changeEveryMapFile2.yml:16]: haha. ok, will re-record something simpler next pass through my PRs.

Comment (pokey) [reply packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/go/changeEveryMapFile2.yml]: Oh yeah, way easier to read, thank you! I find myself quite convinced, and it's useful to know that's how `*` behaves

**Review by pokey — APPROVED** [review]
Looks good!

**Pull request merged**